newrelic/hooks/serverless_azure.py

```diff
@@ -0,0 +1,298 @@
+    # able to access the logic in the __init__ method of the Dispatcher
+    # class with Python (in the Portal, this is not done in Python)
+    if not hasattr(instance, "_nr_running_dispatcher"):
+        instance._nr_running_dispatcher = True
+        instance._nr_cold_start = True
+
+    request = bind_params(*args, **kwargs)
+
+    if not request:
+        return await wrapped(*args, **kwargs)
+
+    # For now, NR only supports HTTP triggers
+    function_id = request.invocation_request.function_id
+
+    binding_type = instance._functions.get_function(function_id).trigger_metadata["type"]
+    if not binding_type.startswith("http"):
+        return await wrapped(*args, **kwargs)
+
+    return await wrapped(*args, **kwargs)
+
+
+async def wrap_dispatcher__run_async_func(wrapped, instance, args, kwargs):
+    from azure.functions.http import HttpRequest
+
+    def bind_params(context, func, args, *_args, **_kwargs):
+        return context, func, args
+
+    context, func, params = bind_params(*args, **kwargs)
+
+    application = application_instance(
+        os.environ.get("NEW_RELIC_APP_NAME", os.environ.get("WEBSITE_SITE_NAME", None)), activate=False
+    )
+    if application and not application.active:
+        application.activate()
+    elif not application:
+        application = application_instance(
+            os.environ.get("NEW_RELIC_APP_NAME", os.environ.get("WEBSITE_SITE_NAME", None))
+        )
+        application.activate()
+
+    http_request = None
+    for key, value in params.items():
+        if isinstance(value, HttpRequest):
+            http_request = value
+            url_split = urlparse.urlsplit(http_request.url)
+            scheme = url_split.scheme
+            query = url_split.query
+            host, port = url_split.netloc, None if (":" not in url_split.netloc) else url_split.netloc.split(":")
+            break
+
+    # If this is an HTTP http_request object, create a web transaction.
+    # Otherwise, create appropriate type of Transaction in the future
+    if http_request:
+        transaction = WebTransaction(
+            application=application,
+            name=context.function_name,
+            group="AzureFunction",
+            scheme=scheme,
+            host=host,
+            port=port,
+            request_method=http_request.method,
+            request_path=http_request.url,
+            query_string=query,
+            headers=dict(http_request.headers),
+            source=func,
+        )
+
+        # For now, only HTTP triggers are supported
+        trigger_type = "http"
+
+        if hasattr(instance, "_nr_cold_start"):
+            cold_start = True
+            # Delete the attribute so that subsequent calls to this
+            # method are noted as not being cold starts
+            del instance._nr_cold_start
+        else:
+            cold_start = False
+
+        website_owner_name = os.environ.get("WEBSITE_OWNER_NAME", None)
```

Comment:
Maybe this logic to grab the subscription ID and resource group name could also be pulled out into a separate method since it seems to be reused in the async wrapper as well